testutil/contract.go

```diff
@@ -0,0 +1,107 @@
+package testutil
+
+import (
+	"errors"
+	"math/big"
+
+	cryptotypes "github.com/cosmos/cosmos-sdk/crypto/types"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	abci "github.com/tendermint/tendermint/abci/types"
+
+	"github.com/ethereum/go-ethereum/common"
+	ethtypes "github.com/ethereum/go-ethereum/core/types"
+	"github.com/ethereum/go-ethereum/crypto"
+
+	"github.com/evmos/evmos/v11/app"
+	"github.com/evmos/evmos/v11/testutil/tx"
+	evm "github.com/evmos/evmos/v11/x/evm/types"
+)
+
+// DeployContract deploys a contract with the provided private key,
+// compiled contract data and constructor arguments
+func DeployContract(
+	ctx sdk.Context,
+	evmosApp *app.Evmos,
+	priv cryptotypes.PrivKey,
+	queryClientEvm evm.QueryClient,
+	contract evm.CompiledContract,
+	constructorArgs ...interface{},
```

Comment:
This should be the ABI and args respectively

Reply:
But don't we need the Bin too to deploy successfully the contract?

Reply:
We can leave it for a later refactor